app/models/shipit/pull_request.rb

```diff
@@ -163,6 +163,8 @@ def merge!
         commit_message: 'Merged by Shipit',
         merge_method: 'merge',
       )
+      stack.undeployed_commits_count += 1
```

Comment:
This way of incrementing a column is heavily subject to race conditions. Rails provides an API for atomic increments of db columns: http://api.rubyonrails.org/classes/ActiveRecord/CounterCache/ClassMethods.html#method-i-increment_counter